packages/language-server/lib/language-service-manager.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * @typedef {import('typescript')} ts
+ * @typedef {import('typescript').IScriptSnapshot} IScriptSnapshot
+ * @typedef {import('typescript').LanguageService} LanguageService
+ */
+
+import { fileURLToPath, pathToFileURL } from 'node:url'
+
+import { createMDXLanguageService } from '@mdx-js/language-service'
+
+import { getDocByFileName } from './documents.js'
+
+/**
+ * @param {ts} ts
+ * @returns {(fileName: string) => IScriptSnapshot | undefined} XXX
+ */
+function createGetScriptSnapshot(ts) {
+  return fileName => {
+    const doc = getDocByFileName(fileName)
+
+    if (doc) {
+      return ts.ScriptSnapshot.fromString(doc.getText())
+    }
+
+    const text = ts.sys.readFile(fileName)
+
+    if (text) {
+      return ts.ScriptSnapshot.fromString(text)
+    }
+  }
+}
+
+/**
+ * @param {string} fileName
+ * @returns {string} The script version
+ */
+function getScriptVersion(fileName) {
+  const doc = getDocByFileName(fileName)
+
+  return doc ? String(doc.version) : '0'
+}
+
+/** @type {LanguageService} */
+let defaultLanguageService
+
+/**
+ * @param {ts} ts
+ * @returns {LanguageService} XXX
```

Comment:
Is this a placeholder for future documentation? Or is there an issue with the return?